web/src/buddy_list.ts

```diff
@@ -263,15 +259,19 @@ export class BuddyList extends BuddyListConf {
         );
     }
 
-    subscribed_participant_ids(): number[] {
-        const {current_sub, pm_ids_set, get_all_participant_ids} = this.render_data;
+    non_participant_users_matching_view_count(): number {
+        const {current_sub, get_all_participant_ids} = this.render_data;
+        // We don't show "participants" for DMs, we just show the
+        // "in this narrow" section (i.e. everyone in the conversation).
+        // Confusingly, we do call this "participants" in the UI.
+        if (current_sub === undefined) {
+            return this.render_data.total_human_subscribers_count;
+        }
         const participant_ids_list = [...get_all_participant_ids()];
-        return participant_ids_list.filter((user_id) => {
-            if (current_sub) {
-                return peer_data.is_user_subscribed(current_sub.stream_id, user_id);
-            }
-            return pm_ids_set.has(user_id);
-        });
+        const subscribed_participant_ids = participant_ids_list.filter((user_id) =>
+            peer_data.is_user_subscribed(current_sub.stream_id, user_id),
+        );
+        return this.render_data.total_human_subscribers_count - subscribed_participant_ids.length;
```

Comment:
I'm not sure this logic is correct -- if you have a bot user, who participated, will this subtract that user from the total here? I don't see anything in the `get_all_participant_ids` or `subscribed_participant_ids` calculation that excludes bots.

Reply:
Ah thanks for calling that out, I can add that to the filter statement here.